packages/vite/src/node/server/sourcemap.ts

```diff
@@ -53,7 +53,7 @@ export async function injectSourcesContent(
           // inject content from source file when sourcesContent is null
           sourceRootPromise ??= computeSourceRoute(map, file)
           const sourceRoot = await sourceRootPromise
-          let resolvedSourcePath = decodeURI(sourcePath)
+          let resolvedSourcePath = fsPathFromUrl(decodeURI(sourcePath))
```

Comment:
```suggestion
          let resolvedSourcePath = cleanUrl(decodeURI(sourcePath))
```
`fsPathFromUrl` does more than stripping query params and hashes. I think [`cleanUrl`](https://github.com/vitejs/vite/blob/6a7dde54e4ac357798d7124894ff71cb36a36985/packages/vite/src/shared/utils.ts#L32) should be used here as I guess that's what you want.

Reply:
Good call, it does. Changed to use that.